felix/bpf-gpl/tc.c

```diff
@@ -1128,17 +1134,18 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct cali_tc_ctx *ctx
 	CALI_DEBUG("Entering calico_tc_skb_accepted\n");
 	struct cali_tc_state *state = ctx->state;
 	bool fib = true;
-	struct ct_create_ctx ct_ctx_nat = {};
+	struct ct_create_ctx *ct_ctx_nat = &ctx->scratch->ct_ctx_nat;
```

Comment:
Does this scratch value get zeroed somewhere? Previously we'd zero it; just worried about leakage between packets.

Reply:
It is memset to zero on line 1283 before the first use. It is only declared globally because the one branch of switch would have to be otherwise indented. So it is zeroed now only in that branch.